### PR TITLE
{bio}[GCCcore/6.4.0] biomart-perl v0.7_e6db561

### PR DESCRIPTION
**Pull request by vanzod**
Requires ~~#6744~~

### CODE REVIEW

**Comment by vanzod** [comment]
Test report by @vanzod
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
cermis - Linux debian 9.4, Intel(R) Xeon(R) CPU           E5620  @ 2.40GHz, Python 2.7.13
See https://gist.github.com/2b759d00dbadb48a2b42e2391c2319fd for a full test report.

File: easybuild/easyconfigs/b/biomart-perl/biomart-perl-0.7_e6db561-GCCcore-6.4.0-Perl-5.26.0.eb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+modextrapaths = {
+    'PERL5LIB': 'lib'
+}
+
+marturlloc = '%(installdir)s/conf/martURLLocation.xml'
+
+postinstallcmds = [
+    'sed -ni "/<MartRegistry>/q;p" %s' % marturlloc,
+    'curl https://useast.ensembl.org/biomart/martservice?type=registry >> %s' % marturlloc,
```
Comment (boegel) [inline easybuild/easyconfigs/b/biomart-perl/biomart-perl-0.7_e6db561-GCCcore-6.4.0-Perl-5.26.0.eb]: @vanzod Is this some kind of mirror? What does this do exactly?

File: easybuild/easyconfigs/b/biomart-perl/biomart-perl-0.7_e6db561-GCCcore-6.4.0-Perl-5.26.0.eb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+]
+
+modloadmsg = "Copy the registry file locally from:\n%(installdir)s/conf/martURLLocation.xml\n\n"
```
Comment (boegel) [inline easybuild/easyconfigs/b/biomart-perl/biomart-perl-0.7_e6db561-GCCcore-6.4.0-Perl-5.26.0.eb]: Locally to where?

File: easybuild/easyconfigs/b/biomart-perl/biomart-perl-0.7_e6db561-GCCcore-6.4.0-Perl-5.26.0.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/b/biomart-perl/biomart-perl-0.7_e6db561-GCCcore-6.4.0-Perl-5.26.0.eb]: Also, why not do this via a patch file?

(we really need better support for runtime patching without relying on `sed`...)

Comment (vanzod) [reply easybuild/easyconfigs/b/biomart-perl/biomart-perl-0.7_e6db561-GCCcore-6.4.0-Perl-5.26.0.eb]: That file must be modified based on the content of that URL. I do not see how to do it from a patch...

File: easybuild/easyconfigs/b/biomart-perl/biomart-perl-0.7_e6db561-GCCcore-6.4.0-Perl-5.26.0.eb (hunk 2 above)
Comment (vanzod) [reply easybuild/easyconfigs/b/biomart-perl/biomart-perl-0.7_e6db561-GCCcore-6.4.0-Perl-5.26.0.eb]: The executon of the BioMart Perl scripts requires this file to identify the source URLs for the data. The data is then cached in the same directory where the file is. Hence it needs to reside in a location where the user has write permission.
I will rephrase it.

File: easybuild/easyconfigs/b/biomart-perl/biomart-perl-0.7_e6db561-GCCcore-6.4.0-Perl-5.26.0.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/b/biomart-perl/biomart-perl-0.7_e6db561-GCCcore-6.4.0-Perl-5.26.0.eb]: Right, good point...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2594.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/70b35481c03fc64d7c50f53459c49e75 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3110.skitty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/418f2ade461a2c03269fcda06a644a3a for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
gligar03.gligar.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/9ce285a5aa828a5b1eaee483f8001c2c for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2594.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/a2c135e1749737171f31fbae1239ef5d for a full test report.

**Comment by boegel** [comment]
Going in, thanks @vanzod!

**Pull request merged**